### PR TITLE
Tweak regex used to extract directives

### PR DESCRIPTION
**Pull request by hayd**
This also allows the : to be optional, for example
```
{{repl
    > foo()
}}
```

### CODE REVIEW

File: src/Directives.jl [hunk 1]
```diff
@@ -21,7 +21,7 @@ parsebracket(:: directive"", text) = ...
 """
 macro directive_str(text) :(Directive{$(quot(symbol(text)))}) end
 
-const DIRECTIVE_MARKER = r"^(\w+):\s*(?s)(.*)"
+const DIRECTIVE_MARKER = r"^(\w+)(:|\s*\n)((?s).*)$"
```
Comment (hayd) [inline src/Directives.jl]: Note: That if directives were to take (optional) arguments they could be grabbed from the second group:

    (:.*|\s*\n)

**Comment by MichaelHatherly** [comment]
Thanks.

**Comment by hayd** [comment]
Another question is about args of parsebracket/directives (as we discussed on other commit).

RE your `reverse` example, it's unclear to me whether reverse should take the arg or the text.

e.g.
```
{{reverse: hello world}}
```
is `parsebrackets(directive"reverse", text="", arg="hello world")`

vs
```
{{reverse:
hello world
}}
```
is `parsebrackets(directive"reverse", text="hello world", arg="")`

---

parsebrackets would fallthrough so that:
```
parsebrackets{D}(directive{D}, text="hello world", arg="")
parsebrackets{D}(directive{D}, text="hello world")
```
that way if only the latter would need to be defined if no arg was take.

---

Does that make any sense?

**Comment by MichaelHatherly** [comment]
I guess ``arg`` and ``text`` aren't really that different from each other, just the first and second arguments. We could have some kind of argument separator. Or just leave it up to each directive to do whatever it wants to with the raw text, ie. just pass it everything after the ``:``. Not sure what would be more useful.

**Comment by hayd** [comment]
I kindof like using the : only if you want to pass an arg, otherwise it's all text. This is a little tricky in the reverse example: is it text or arg; should we favour it being on a newline (I think so).

**Pull request merged**